service/arm_templates/main.json

```diff
@@ -163,5 +188,63 @@
         }
       }
     },
+    {
+      "apiVersion": "2016-09-01",
+      "name": "master_availability_sets_setup",
+      "type": "Microsoft.Resources/deployments",
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[variables('availabilitySetsTemplateURI')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "clusterID": {"value": "[parameters('clusterID')]"},
+          "prefix": {"value": "[variables('masterPrefix')]"}
+        }
+      }
+    },
+    {
+      "apiVersion": "2016-09-01",
+      "name": "worker_availability_sets_setup",
+      "type": "Microsoft.Resources/deployments",
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[variables('availabilitySetsTemplateURI')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "clusterID": {"value": "[parameters('clusterID')]"},
+          "prefix": {"value": "[variables('workerPrefix')]"}
+        }
+      }
+    },
+    {
+      "apiVersion": "2016-09-01",
+      "name": "[concat('master_nodes_setup_', copyIndex(1))]",
+      "type": "Microsoft.Resources/deployments",
+      "copy": {
+        "name": "master_nodes_setup_copy",
+        "count":  "[length(parameters('mastersCustomConfig'))]"
+      },
+      "properties": {
+        "mode": "incremental",
+        "templateLink": {
+          "uri": "[variables('nodesSetupTemplateURI')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "clusterID": {"value": "[parameters('clusterID')]"},
+          "storageAccountName": {"value": "[reference('cluster_setup').outputs.storageAccountName.value]"},
+          "prefix": {"value": "[concat(variables('masterPrefix'), '-', copyIndex(1))]"},
+          "subnetID": {"value": "[reference('virtual_network_setup').outputs.masterSubnetID.value]"},
+          "availabilitySetID": {"value": "[reference('master_availability_sets_setup').outputs.availabilitySetID.value]"},
+          "disablePasswordAuth": {"value": "[parameters('masterDisablePasswordAuth')]"},
+          "nodeCustomConfig": {"value": "[parameters('mastersCustomConfig')[copyIndex()]]"},
```

Comment:
I think it would be great to pass all the parameters from `nodeCustomConfig` explicitly here, and take them as separate variables in `nodes_setup.json`. 

Benefits:

* We can describe separate variables in `nodes_setup.json`, and add comments.
* `nodeCustomConfig` properties are accessed in a single place - then it's easy to discover what parameters it ha.
*  The object parameter stays in scope of a single file/template - again, it's easier to discover the object definition.
* `nodes_setup.json` becomes easier to understand, better documented on its own.